client/components/login/login.js

```diff
@@ -206,27 +195,25 @@ export default class Login extends React.Component {
       }),
     })
       .then((res = {}) => {
-        if (res.data) setIsActive(res.data.is_active);
-        return handleAuthentication(
-          settings.mobile_phone_verification,
-          res.data,
-        );
+        return handleAuthentication(res.data);
       })
       .catch((error) => {
         const {data} = error.response;
-        setIsActive(data.is_active);
         if (
           error.response.status === 401 &&
           settings.mobile_phone_verification &&
           data.is_active
         ) {
-          return handleAuthentication(true);
+          return handleAuthentication(data);
         }
+        userData.is_active = false;
+        setUserData(userData);
```

Comment:
why can't we simply call this earlier and avoid having it both here and in `handleAuthentication`?